tests/unit/test-class-wc-payment-gateway-wcpay.php

```diff
@@ -1065,6 +1073,50 @@ public function test_full_level3_data_with_fee() {
 		$this->assertEquals( $expected_data, $level_3_data );
 	}
 
+	public function test_full_level3_data_with_fee_and_negative_price_product() {
+		$expected_data = [
+			'merchant_reference'   => '210',
+			'customer_reference'   => '210',
+			'shipping_amount'      => 3800,
+			'line_items'           => [
+				(object) [
+					'product_code'        => 30,
+					'product_description' => 'Beanie with Logo',
+					'unit_cost'           => 1800,
+					'quantity'            => 1,
+					'tax_amount'          => 270,
+					'discount_amount'     => 0,
+				],
+				(object) [
+					'product_code'        => 'fee',
+					'product_description' => 'fee',
+					'unit_cost'           => 1000,
+					'quantity'            => 1,
+					'tax_amount'          => 150,
+					'discount_amount'     => 0,
+				],
+				(object) [
+					'product_code'        => 42,
+					'product_description' => 'Negative Product Price',
+					'unit_cost'           => 0,
+					'quantity'            => 1,
+					'tax_amount'          => 270,
+					'discount_amount'     => 1899,
+				],
+			],
+			'shipping_address_zip' => '98012',
+			'shipping_from_zip'    => '94110',
+		];
+
+		update_option( 'woocommerce_store_postcode', '94110' );
+
+		$this->mock_wcpay_account->method( 'get_account_country' )->willReturn( 'US' );
+		$mock_order   = $this->mock_level_3_order( '98012', true, true, 1, 1 );
```

Comment:
Can we omit the fee inclusion here: `$mock_order = $this->mock_level_3_order( '98012', false, true, 1, 1 );`?
That would make test intention clearer.

Reply:
Addressed in [5c3d486](https://github.com/Automattic/woocommerce-payments/pull/4244/commits/5c3d4866fba7668abdf766121e212dbc94eb87f7)